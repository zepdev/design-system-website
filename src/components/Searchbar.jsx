import React from 'react'
import SearchIcon from './icons/SearchIcon'

const Searchbar = () => (
  <div className="zds-searchbar">
    <SearchIcon className="zds-searchbar--icon" />
    <input
      type="text"
      className="zds-searchbar--input"
      placeholder="Type here to search"
    />
  </div>
)

export default Searchbar
