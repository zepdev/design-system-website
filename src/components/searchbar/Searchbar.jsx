import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import SearchIcon from '../icons/SearchIcon'

const styles = theme => ({
  root: {
    height: '100%',
    background: 'transparent',
    display: 'flex',
    alignItems: 'center,',
  },
  icon: {
    width: 30,
    height: 30,
    color: theme.colors.white,
  },
  input: {
    background: theme.colors.white,
    color: theme.colors.grey4,
    fontSize: '0.875rem',
    width: '100%',
    border: 'none',
    display: 'none',
  },
  [`@media (min-width: ${ theme.breakpoints.m })`]: {
    root: {
      flexBasis: '80%',
      background: theme.colors.white,
      padding: '0.75rem 1.5rem',
    },
    icon: {
      width: 24,
      height: 24,
      color: theme.colors.grey3,
    },
    input: {
      display: 'block',
    },
  },
})

const Searchbar = ({ classes }) => (
  <div className={classes.root}>
    <SearchIcon className={classes.icon} />
    <input type="text" className={classes.input} placeholder="Type here to search" />
  </div>
)

Searchbar.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Searchbar)
