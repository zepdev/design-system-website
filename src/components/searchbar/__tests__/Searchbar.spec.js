import React from 'react'
import { render } from 'react-testing-library'
import Searchbar from '../Searchbar'

describe('Searchbar', () => {
  it('renders correctly', () => {
    const component = render(<Searchbar />)
    expect(component).toMatchSnapshot()
  })
})
