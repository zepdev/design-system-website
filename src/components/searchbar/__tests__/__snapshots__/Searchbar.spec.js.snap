// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Searchbar-root-0-1-1"
      >
        <button
          class="Searchbar-button-0-1-2 zep-button"
        >
          <svg
            aria-labelledby="title"
            class="Searchbar-icon-0-1-3"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_search"
            >
              Search Icon
            </title>
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              fill="currentColor"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
        <input
          class="Searchbar-input-0-1-4"
          placeholder="Type here to search"
          type="text"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Searchbar-root-0-1-1"
    >
      <button
        class="Searchbar-button-0-1-2 zep-button"
      >
        <svg
          aria-labelledby="title"
          class="Searchbar-icon-0-1-3"
          height="32"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title_search"
          >
            Search Icon
          </title>
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            fill="currentColor"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </button>
      <input
        class="Searchbar-input-0-1-4"
        placeholder="Type here to search"
        type="text"
      />
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
